src/main/java/org/generationcp/middleware/dao/ims/TransactionDAO.java

```diff
@@ -741,8 +743,24 @@ private String formatDate(final Date date) {
 		try {
 			return TransactionDto.class.getConstructor(Integer.class, String.class, String.class,  Double.class,
 			 String.class, Date.class, Integer.class, Integer.class, String.class, String.class, Integer.class, String.class, String.class);
-		} catch (NoSuchMethodException ex) {
+		} catch (final NoSuchMethodException ex) {
 			throw new RuntimeException(ex);
 		}
 	}
+
+
+	public Transaction saveTransaction(final TransactionDto transactionDto, final Lot lot, final TransactionStatus transactionStatus) {
+		final Transaction transaction = new Transaction();
```

Comment:
@vmaletta Lot should not be a parameter of this function if TransactionDto contains a Lot with the id. 
There is no need to query for the lot, just build a new Lot entity and set the id

Reply:
@vmaletta if TransactionDto has a type, then the status can be calculated and not passed as a parameter to the function because it can cause some inconsistencies.